### DISCUSSION TITLE
`x-for` runs twice on Alpine CDN 3.14.4

### DISCUSSION THREAD

**Discussion by monotykamary** [alpinejs/alpine#3997] [7. Bugs (archived)] [answered]
I help to run and manage https://note.d.foundation/, one of front-facing internal notes app. I noticed that when upgrading to `3.14.4`, all of the templates that run `x-for` runs twice. e.g: 

**Code:**
```html
<nav class="tab-menu" x-data="{
    currentUrl: `{{ .RelPermalink }}`,
    links: [
      { title: 'Home', url: '/' },
      { title: 'Consulting', url: '/consulting/' },
      { title: 'Labs', url: '/labs/' },
      { title: 'Earn', url: '/earn/' },
      { title: 'Hiring', url: '/hiring/' },
      { title: 'Newsletter', url: '/newsletter/' },
      { title: 'Events', url: '/events/' },
    ],
    isActiveUrl(url) {
      const linkUrlRoot = url.split('/')[1];
      const currentUrlRoot = this.currentUrl.split('/')[1];
      return linkUrlRoot === currentUrlRoot;
    },
  }">
  <ul>
    <template x-for="item in links">
      <li x-bind:class="{ 'active': isActiveUrl(item.url) }">
        <a x-bind:href="item.url" x-text="item.title"></a>
      </li>
    </template>
  </ul>
</nav>
```

**Before:**
```
<script defer src="https://cdn.jsdelivr.net/npm/alpinejs@3.13.3/dist/cdn.min.js"></script>
```
<img width="917" alt="image" src="https://github.com/alpinejs/alpine/assets/1130103/3348c085-a7da-4afc-85f1-c111672c3c30">

**After:**
```
<script defer src="https://cdn.jsdelivr.net/npm/alpinejs@3.13.4/dist/cdn.min.js"></script>
```
<img width="881" alt="image" src="https://github.com/alpinejs/alpine/assets/1130103/8aa5b455-da83-41c1-8b47-44010f061447">

The problem is very consistent, although I'm not sure if there is an update needed for calling Alpine.js through a CDN.

**Comment by calebporzio** [comment]
![CleanShot 2024-01-24 at 11 55 31@2x](https://github.com/alpinejs/alpine/assets/3670578/46838ba1-050b-4c19-a738-83f3a89bdd68)

Hmm I pasted your code in and it looks fine. Could it be something else?

**Comment by ekwoka** [comment, marked as answer]
Can you retry with 3.13.5?

**Reply by monotykamary** [reply]
Just retried it with 3.13.5 and it works great 🚀

**Comment by ekwoka** [comment]
Also, just fyi, you don't need your URL logic.

you can just use `:local-link` https://developer.mozilla.org/en-US/docs/Web/CSS/:local-link instead of your fancy url split stuff...

**Reply by calebporzio** [reply]
Can he though? Looks like it's not supported 🤔 

![CleanShot 2024-01-25 at 08 18 06@2x](https://github.com/alpinejs/alpine/assets/3670578/5ccc14c9-b2eb-4f27-82c6-58d8505ac3c3)

**Reply by ekwoka** [reply]
oh dang.

I though it already existed :(
